File: 0684/sesión25 - Reportes/Front-End/SoftVen/SoftVenBusiness/ProductoBO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Web;
using SoftVenBusiness.productoWS;

namespace SoftVen.BO
{
    public class ProductoBO
    {
        private ProductosClient productosClienteSOAP;

        public ProductoBO()
        {
            this.productosClienteSOAP = new ProductosClient();
        }

        public int Insertar(string nombre, string descripcion, double precio)
        {
            return this.productosClienteSOAP.insertar(nombre, descripcion, precio);
        }

        public int Modificar(int productoId, string nombre, string descripcion, double precio)
        {
            return this.productosClienteSOAP.modificar(productoId, nombre, descripcion, precio);
        }

        public int Eliminar(int productoId)
        {
            return this.productosClienteSOAP.eliminar(productoId);
        }

        public productosDTO ObtenerPorId(int productoId)
        {
            return this.productosClienteSOAP.obtenerPorId(productoId);
        }

        public BindingList<productosDTO> ListarTodos()
        {
            productosDTO[] arreglo = this.productosClienteSOAP.listarTodos();
            return new BindingList<productosDTO>(arreglo);
        }

        public byte[] reporteVendedores()
        {
            return this.productosClienteSOAP.reporteVendedores();
        }

        public byte[] reporteTiendas(int pTiendaId)
        {
            return this.productosClienteSOAP.reporteTiendas(pTiendaId);
        }

        public void abrirReporte(HttpResponse response, string nombreDeReporte, byte[] reporte)
        {
            response.Clear();
            response.ContentType = "application/pdf";
            response.AddHeader("Content-Disposition", "inline;filename=" + nombreDeReporte + ".pdf");
            response.BinaryWrite(reporte);
            response.End();
        }
    }
}

File: 0684/sesión22 - WS - RESTFul/Back-End/ServidorRESTFul/SoftVenBusiness/src/main/java/pe/edu/pucp/softven/bo/server/ProductoBO.java
package pe.edu.pucp.softven.bo.server;

import java.util.ArrayList;
import pe.edu.pucp.softven.dao.ProductoDAO;
import pe.edu.pucp.softven.daoImp.ProductoDAOImpl;
import pe.edu.pucp.softven.model.ProductosDTO;

public class ProductoBO {

    private ProductoDAO productoDAO;
    
    public ProductoBO(){
        this.productoDAO = new ProductoDAOImpl();
    }
    
    public ProductosDTO insertar(ProductosDTO productosDTO){
        Integer id = this.productoDAO.insertar(productosDTO);
        productosDTO.setProductoId(id);
        return productosDTO;
    }
    
    public ProductosDTO obtenerPorId(Integer productoId){
        return this.productoDAO.obtenerPorId(productoId);
    }
    
    public ArrayList<ProductosDTO> listarTodos(){
        return this.productoDAO.listarTodos();
    }
    
    public Integer modificar(ProductosDTO productosDTO){        
        return this.productoDAO.modificar(productosDTO);
    }
    
    public Integer eliminar(Integer productoId){
        ProductosDTO productosDTO = new ProductosDTO();
        productosDTO.setProductoId(productoId);
        return this.productoDAO.eliminar(productosDTO);
    }
    
}
